node/src/blockchain/blockchain_interface_utils.rs

```diff
@@ -0,0 +1,1412 @@
+        let timestamp_before = SystemTime::now();
+
+        let result = send_payables_within_batch(
+            logger,
+            chain,
+            web3_batch,
+            consuming_wallet.clone(),
+            gas_price.clone(),
+            pending_nonce,
+            new_fingerprints_recipient,
+            accounts.clone(),
+        )
+        .wait();
+
+        System::current().stop();
+        system.run();
+        let tlh = TestLogHandler::new();
+        let timestamp_after = SystemTime::now();
+        let recording_result = accountant_recording.lock().unwrap();
+        let processed_payments = result.unwrap();
+        let message = recording_result.get_record::<PendingPayableFingerprintSeeds>(0);
+        assert_eq!(recording_result.len(), 1);
+        assert!(timestamp_before <= message.batch_wide_timestamp);
+        assert!(timestamp_after >= message.batch_wide_timestamp);
+        assert_eq!(
+            message.hashes_and_balances,
+            vec![
+                HashAndAmount {
+                    hash: H256::from_str(
+                        "35f42b260f090a559e8b456718d9c91a9da0f234ed0a129b9d5c4813b6615af4"
+                    )
+                    .unwrap(),
+                    amount: accounts_1.balance_wei
+                },
+                HashAndAmount {
+                    hash: H256::from_str(
+                        "7f3221109e4f1de8ba1f7cd358aab340ecca872a1456cb1b4f59ca33d3e22ee3"
+                    )
+                    .unwrap(),
+                    amount: accounts_2.balance_wei
+                },
+            ]
+        );
+        assert_eq!(
+            processed_payments[0],
+            ProcessedPayableFallible::Correct(PendingPayable {
+                recipient_wallet: accounts_1.wallet,
+                hash: H256::from_str(
+                    "35f42b260f090a559e8b456718d9c91a9da0f234ed0a129b9d5c4813b6615af4"
+                )
+                .unwrap()
+            })
+        );
+        assert_eq!(
+            processed_payments[1],
+            ProcessedPayableFallible::Correct(PendingPayable {
+                recipient_wallet: accounts_2.wallet,
+                hash: H256::from_str(
+                    "7f3221109e4f1de8ba1f7cd358aab340ecca872a1456cb1b4f59ca33d3e22ee3"
+                )
+                .unwrap()
+            })
+        );
+        tlh.exists_log_containing(
+            &format!("DEBUG: {test_name}: Common attributes of payables to be transacted: sender wallet: {}, contract: {:?}, chain_id: {}, gas_price: {}",
+                     consuming_wallet,
+                     chain.rec().contract,
+                     chain.rec().num_chain_id,
+                     gas_price
+            )
+        );
+        tlh.exists_log_containing(&format!(
+            "INFO: {test_name}: {}",
+            transmission_log(chain, &accounts, gas_price)
+        ));
+    }
+
+    #[test]
+    fn send_payables_within_batch_all_payments_fail() {
```

Comment:
Oh guys... I'm again loosing my strength as there's so much I could comment...

This is an aggregated comment, more about principles that are valid in general. 

Imagine we have a happy-path test called A about a certain function, we also have a test with one error case called B, and another error tested under C. All are basically the same act. Then it stands that:
 
1) The test A tends to contain the biggest portion of assertions. It's a good habit that the remaining tests don't repeat the assertions unless they are somehow important to the specific test case. It usually means a saving of a huge bunch of duplicated lines. 
With B and C, concentrate solely on the assertions that assess values that sourced somehow from the error. 

2) If these tests require lots of code to set them up before we can fire them, it's advisable to look around for a possibility how the common parts could be refactored into outboards methods. I strongly recommend it in these three related tests about `send_payables_within_batch()` too. These contain an awful amount of common code.


And even more generalized advise: despite you're dealing with code partially written and you need to complete and refactor it, you still are obliged to try checking for the test driven paradigm not broken. 
It can be done, but it's tedious. 
In most of the tests in this file, it's no effort to figure out they barely were written to test-drive the code, instead they suffer from an attempt to provide a reasonable test coverage, but it was often even overstressed or underrated.